Update Helm 3 doc to improve usability

hickeyma opened the pull request:
I was using the [Helm3 support](https://github.com/istio/istio/blob/master/manifests/charts/README-helm3.md) doc to install Istio into a cluster with Helm. The doc is good but I found a few things when doing it. I have therefore added some suggestions for improvement which could help someone new to Istio, to better navigate the process.

[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

hickeyma committed:
Update Helm 3 doc to improve usability

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

istio-testing commented:
Hi @hickeyma. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

howardjohn commented on manifests/charts/README-helm3.md:
````diff
@@ -1,57 +1,64 @@
-```shell script
- helm3 install -n istio-system istio-17 manifests/charts/istio-control/istio-discovery
+```console
+ helm install -n istio-system istio-17 manifests/charts/istio-control/istio-discovery -f manifests/charts/global.yaml
````
I thought we removed the requirement for this. @jacob-delgado is it still required, and if not can it be removed?

morvencao replied:
It should be removed after https://github.com/istio/istio/pull/25186

hickeyma replied:
That is a typo by me, apologies. It is working without the global values.

hickeyma replied:
Removed.

shamsher31 commented on manifests/charts/README-helm3.md:
```diff
@@ -1,57 +1,64 @@
 
-The manifests/ templates support both helm2 and helm3. Please do not introduce helm3-specific changes, many
-users are still using helm2 and the operator is currently using the helm2 code to generate.
+The Helm charts are support both by Helm v2 and Helm v3. Please do not introduce Helm v3 specific changes as many
```
```suggestion
The Helm charts are supported both by Helm v2 and Helm v3. Please do not introduce Helm v3 specific changes as many
```

hickeyma replied:
Updated

shamsher31 commented on manifests/charts/README-helm3.md:
```diff
@@ -1,57 +1,64 @@
 
-- 'base' creates cluster-wide CRDs, cluster bindings, cluster resources and the istio-system namespace.
-  It is possible to customize the namespace, but not recommended.
+- 'base' creates cluster-wide CRDs, cluster bindings and cluster resources. It is possible to change the namespace from `istio-system` but it is not recommended.
```
```suggestion
- `base` creates cluster-wide CRDs, cluster bindings, and cluster resources. It is possible to change the namespace from the `istio-system` but it is not recommended.
```

hickeyma replied:
Done

shamsher31 commented on manifests/charts/README-helm3.md:
````diff
@@ -1,57 +1,64 @@
 ```
 
-- 'ingress' to install a Gateway
+- 'gateways' install a load balancer with `ingress` and `egress`. You can install it multiple times with different revisions but they must be installed in separate namespaces.
````
```suggestion
- `gateways` install a load balancer with `ingress` and `egress`. You can install it multiple times with different revisions but they must be installed in separate namespaces.
```

hickeyma replied:
Done

shamsher31 commented on manifests/charts/README-helm3.md:
````diff
@@ -1,57 +1,64 @@
+helm install istio-base -n istio-system manifests/charts/base
 ```
 
 - 'istio-control/istio-discovery' installs a revision of istiod.  You can install it multiple times, with different revisions.
````
```suggestion
- `istio-control/istio-discovery` installs a revision of `istiod`.  You can install it multiple times, with different revisions.
```

hickeyma replied:
Done

shamsher31 commented:
/ok-to-test

hickeyma commented:
I will do an update on the CLA and reviews asap.

hickeyma committed:
Update after reviews

Review comments:
- https://github.com/istio/istio/pull/27598#pullrequestreview-497853977
- https://github.com/istio/istio/pull/27598#pullrequestreview-498224470

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
Thanks for the reviews @howardjohn @morvencao @shamsher31. Updated and ready for review again.

sdake commented:
/ok-to-test

istio-testing commented:
@hickeyma: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 0715b84c2b3f8f47bfe04d76af88b61194be84e3 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/27598/unit-tests_istio/21411) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

sdake commented:
/test unit-tests_istio

istio-testing commented:
@hickeyma: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 0715b84c2b3f8f47bfe04d76af88b61194be84e3 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/27598/unit-tests_istio/21412) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

hickeyma commented:
/retest
